packages/auth/src/accessToken/routes.ts

```diff
@@ -64,26 +66,25 @@ async function introspectToken(
   const introspectionResult = await deps.accessTokenService.introspect(
     body['access_token']
   )
-  if (introspectionResult) {
-    ctx.body = introspectionToBody(introspectionResult)
-  } else {
-    ctx.throw(404, {
-      error: 'invalid_request',
-      message: 'token not found'
-    })
-  }
+  ctx.body = introspectionResult
+    ? introspectionToBody(introspectionResult)
+    : {
+        active: false
+      }
```

Comment:
Maybe we can move this ternary into `introspectionToBody` as well, so that we are a bit more safe with the types? (i.e. have `introspectionToBody` return just `TokenInfo`)